src/main/java/bio/terra/pdao/PrimaryDataAccess.java

```diff
@@ -0,0 +1,52 @@
+package bio.terra.pdao;
+
+import bio.terra.metadata.Study;
+
+/**
+ * In the long term, we want to make the primary data store be pluggable, supporting different implementations.
+ * This interface documents what that pluggable interface might look like. Of course, we never know until we
+ *
+ * We use the term "container" for the object that contains a set of tables. In BigQuery, this would be a
+ * BigQuery dataset. In Postgresql, it would probably be a schema.
+ *
+ * The expected sequence of calls to ensure idempotence:
+ * <ul>
+ *     <li>do sequence</li>
+ *     <ol>
+ *         <li>studyExists - validate that the study does not exist before calling create</li>
+ *         <li>createStudy - if it finds the study, it deletes it and recreates. That covers the case where the
+ *         create operation is interrupted by a service failure and restarts.</li>
+ *     </ol>
+ *     <li>undo</li>
+ *     <ol>
+ *     <li>deleteStudy - called on undo; does not fail if the study has already been deleted</li>
+ *     </ol>
+ * </ul>
+ *
+ * REVIEWERS: Is it worth catching underlying implementation exceptions and packaging them into a PDAO exception?
+ * Seems of marginal value to me, but has a nice aesthetic.
```

Comment:
I like this pattern, prevents users of the PDAO from having to know too much about the inner workings of the PDAO.

Reply:
Implemented and committed